### ISSUE TITLE
[ui] RESOURCE_INIT_FAILURE steps not moved to Errored in side panel

### ISSUE THREAD

**Issue by alangenfeld** [dagster-io/dagster#18368]
### Dagster version

latest

### What's the issue?

If a step fails due to `RESOURCE_INIT_FAILURE`, the step is still listed in the `Preparing` section of the top right section of the runs page instead of being put in to `Errored` 

### What did you expect to happen?

It should move to the `Errored` section

### How to reproduce?

load this debug file with `dagster-webserver-debug`
[a870b539-18a4-4bd0-a707-8c2540ecb3e3.gz](https://github.com/dagster-io/dagster/files/13503558/a870b539-18a4-4bd0-a707-8c2540ecb3e3.gz)

### Deployment type

None

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by alangenfeld** [comment]
![Screenshot 2023-11-29 at 11 15 11 AM](https://github.com/dagster-io/dagster/assets/202219/1f02b8ce-87fa-40d8-8e2f-6c9a316bcd9a)

**Comment by smackesey** [comment]
cc @hellendag this was reported by an enterprise user so we should prioritize a fix

**Comment by alexbuyn** [comment]
We also have this issue @hellendag @smackesey any update on this?

**Comment by smackesey** [comment]
@alexbuyn I don't believe this has been fixed yet-- I've put it into an internal triage queue. @hellendag can you confirm no movement on this?
